lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -171,6 +171,27 @@ LogicalResult SubgraphOp::verify() {
   return success();
 }
 
+LogicalResult SubgraphOp::inferReturnTypeComponents(
+    MLIRContext * /*context*/, ::std::optional<Location> /*location*/,
+    ValueShapeRange /*operands*/, DictionaryAttr /*attributes*/,
+    OpaqueProperties /*properties*/, RegionRange regions,
+    SmallVectorImpl<ShapedTypeComponents> &inferredReturnShapes) {
+
+  llvm::SmallVector<ShapedTypeComponents, 2> returnShapes;
+
+  Operation *terminator = regions.front()->front().getTerminator();
```

Comment:
nit: since we know we are in a SubgraphOp we can use `getEnclaveBody().getTerminator()`. We won't have more than one region. This would work right?

Reply:
We don't have access to the original operation in this inferReturnTypeComponents and instead we have the operation components like region, operands, ... .